include/swift/SIL/SILConstants.h

```diff
@@ -313,6 +336,32 @@ class SymbolicValue {
 
   SymbolicValue getEnumPayloadValue() const;
 
+
+  /// Return a symbolic value that represents the address of a memory object.
+  static SymbolicValue getAddress(unsigned objectID) {
+    SymbolicValue result;
+    result.representationKind = RK_DirectAddress;
+    result.aux.directAddress_objectID = objectID;
+    return result;
+  }
+
+  /// Return a symbolic value that represents the address of a memory object
+  /// indexed by a path.
+  static SymbolicValue getAddress(unsigned objectID, ArrayRef<unsigned> indices,
+                                  llvm::BumpPtrAllocator &allocator);
+
+  /// Return the object ID of an address value.
+  unsigned getAddressValueObjectID() const {
+    if (representationKind == RK_DirectAddress)
+      return aux.directAddress_objectID;
+    assert(representationKind == RK_DerivedAddress);
+    return value.derivedAddress[0];
```

Comment:
check (via `derivedAddress_numEntries`) if `derivedAddress[0]` is valid?

Reply:
It is always valid.  I'll add an assertion to check that.  Thanks!